agent/api/task/task.go

```diff
@@ -562,6 +586,83 @@ func (task *Task) initializeEFSVolumes(cfg *config.Config, dockerClient dockerap
 	return nil
 }
 
+func buildVolumeNameForExecAgentBinary(binaryName string) string {
+	return fmt.Sprintf("%s-%s", internalExecAgentNamePrefix, binaryName)
+}
+
+// initializeExecAgentResources specifies the necessary volumes and mount points in all of the task containers in order for the
+// exec agent to run upon container start.
+func (task *Task) initializeExecAgentResources(cfg *config.Config) error {
+	if !task.IsExecAgentEnabled() {
+		return nil
+	}
+
+	tId, err := task.GetID()
+	if err != nil {
+		return err
+	}
+
+	execAgentBinNames := []string{execAgentBinName, execAgentSessionWorkerBinName, execAgentSessionLoggerBinName}
+
+	// Append exec agent binaries to task volumes
+	for _, bn := range execAgentBinNames {
+		task.Volumes = append(task.Volumes,
+			TaskVolume{
+				Type: HostVolumeType,
+				Name: buildVolumeNameForExecAgentBinary(bn),
+				Volume: &taskresourcevolume.FSHostVolume{
+					FSSourcePath: filepath.Join(execAgentHostBinDir, bn),
+				},
+			})
+	}
+
+	// Append certificates volume
+	task.Volumes = append(task.Volumes,
+		TaskVolume{
+			Type: HostVolumeType,
+			Name: internalExecAgentCertVolumeName,
+			Volume: &taskresourcevolume.FSHostVolume{
+				FSSourcePath: execAgentHostCertFile,
+			},
+		})
+
+	// Add log volumes and mount points to all containers in this task
+	for _, c := range task.Containers {
+		logSuffix := fmt.Sprintf("%s-%s", tId, c.Name)
```

Comment:
I believe this creates log dirs roughly as - 
```
../exec/taskID1-container1 
../exec/taskID1-container2 
../exec/taskID2-container1 
```

instead why not just create a parent dir for task and have container logs under it? 

something like:
```
../exec/taskID1/container1 
../exec/taskID1/container2 
../exec/taskID2/container1 
```

Reply:
I'd be OK with that. I'd like to hear Ray's opinion on this as well since this was a deviation from the spec in the design doc.

Reply:
The design doc spec assumed that we would have container id at the time of the bind mount, which as you Angel pointed out would not exist (oops!) at the time of the bind mount.
I like the idea of setting the containers for a task in their own directory.  This gives us the unique host-level log file name and a neat organization scheme.  I think it does make these logs more manageable.
(And this also might make cleaning up logs later simpler, based on just the taskID.)

Reply:
Container name allows hyphen https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_name. Just wondering is the path valid if the name starts with hyphen?

Reply:
Directories can't start with hyphens. I guess I can prepend container- to the container name?
e.g. 
two containers: `abc` and `-other`
../exec/taskId1/container-abc
../exec/taskId1/container--other

thoughts?

Reply:
Alternatively I could remove any trailing hyphens, but there's the chance that the container name is just `-`, in which case I'd end up with a blank name.

Reply:
Fixed: Creating sub directories now. For the container name subdirectory, I am trimming leading hyphens. If the container only has hyphens in the name I fallback to `nameless-continer-UUID`, since it's not possible to create directories with leading hyphens in the name.